dashboard/app/controllers/user_school_infos_controller.rb

```diff
@@ -0,0 +1,63 @@
+class UserSchoolInfosController < ApplicationController
+  #before_action :authenticate_user!
+
+  # PATCH /api/v1/users_school_infos/<id>/update_last_confirmation_date
+  def update_last_confirmation_date
+    user_school_info = UserSchoolInfo.find(params[:id])
+    result = user_school_info.update(last_confirmation_date: DateTime.now)
+    if result
```

Comment:
I don't think this will do what you expect it to do; the return value of `update` is always going to be the object itself, whether it passes or not. `result.valid?` might be what you want instead

Reply:
Refactored method.

Reply:
@nkiruka can you share the commit hash for the refactor? I think this might still have the same issue in the latest change set.

> The resulting object is returned whether the object was saved successfully to the database or not.

The `update!` version should throw, but then I don't think the `else` case will ever be hit?